### DISCUSSION TITLE
Cryptomator successfully downloaded but failed the hash check, why?

### DISCUSSION THREAD

**Discussion by Xelbayria** [ScoopInstaller/Extras#11064] [Q&A]
I believe this is not a bug but just a problem on my side. Do anyone know have any ideas as for whats going on? I also tried `cmd.exe`, same result. It is saying it cannot find the file to check against the hash via the manifest?
I've tested this on my laptop, it's fine without any problem... however, on my computer, there is one problem. I don't know what caused it: 

```shell
C:\Users> scoop download cryptomator

INFO  Starting download for cryptomator...
Cryptomator-1.7.5-x64.msi (47.4 MB) [=========================================================================] 100%
Move-Item: C:\Program Scoop\apps\scoop\current\lib\install.ps1:89
Line |
  89 |          Move-Item "$cached.download" $cached -force
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Access to the path is denied.
Checking hash of Cryptomator-1.7.5-x64.msi ... Get-FileHash: C:\Program Scoop\apps\scoop\current\lib\install.ps1:674
Line |
 674 |      $actual = (Get-FileHash -Path $file -Algorithm $algorithm).Hash.T …
     |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot find path 'C:\Program
     | Scoop\cache\cryptomator#1.7.5#https_github.com_cryptomator_cryptomator_releases_download_1.7.5_Cryptomator-1.7.5-x64.msi' because it does not exist.
InvalidOperation: C:\Program Scoop\apps\scoop\current\lib\install.ps1:674
Line |
 674 |      $actual = (Get-FileHash -Path $file -Algorithm $algorithm).Hash.T …
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | You cannot call a method on a null-valued expression.
ERROR Hash check failed!
App:         extras/cryptomator
URL:         https://github.com/cryptomator/cryptomator/releases/download/1.7.5/Cryptomator-1.7.5-x64.msi
Expected:    0f8f14174967dcd804f9311cdc684cab8c91e1f84e3f99c89edee8e44880e030
Actual:
ERROR
Please try again or create a new issue by using the following link and paste your console output:
https://github.com/ScoopInstaller/Extras/issues/new?title=cryptomator%401.7.5%3a+hash+check+failed

'cryptomator' (1.7.5) was downloaded successfully!

```

**Comment by ivaquero** [comment]
> I believe this is not a bug but just a problem on my side. Do anyone know have any ideas as for whats going on? I also tried `cmd.exe`, same result. It is saying it cannot find the file to check against the hash via the manifest?
> 
> I've tested this on my laptop, it's fine without any problem... however, on my computer, there is one problem. I don't know what caused it: 
> 
> 
> 
> ```shell
> 
> C:\Users> scoop download cryptomator
> 
> 
> 
> INFO  Starting download for cryptomator...
> 
> Cryptomator-1.7.5-x64.msi (47.4 MB) [=========================================================================] 100%
> 
> Move-Item: C:\Program Scoop\apps\scoop\current\lib\install.ps1:89
> 
> Line |
> 
>   89 |          Move-Item "$cached.download" $cached -force
> 
>      |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
>      | Access to the path is denied.
> 
> Checking hash of Cryptomator-1.7.5-x64.msi ... Get-FileHash: C:\Program Scoop\apps\scoop\current\lib\install.ps1:674
> 
> Line |
> 
>  674 |      $actual = (Get-FileHash -Path $file -Algorithm $algorithm).Hash.T …
> 
>      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
>      | Cannot find path 'C:\Program
> 
>      | Scoop\cache\cryptomator#1.7.5#https_github.com_cryptomator_cryptomator_releases_download_1.7.5_Cryptomator-1.7.5-x64.msi' because it does not exist.
> 
> InvalidOperation: C:\Program Scoop\apps\scoop\current\lib\install.ps1:674
> 
> Line |
> 
>  674 |      $actual = (Get-FileHash -Path $file -Algorithm $algorithm).Hash.T …
> 
>      |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
>      | You cannot call a method on a null-valued expression.
> 
> ERROR Hash check failed!
> 
> App:         extras/cryptomator
> 
> URL:         https://github.com/cryptomator/cryptomator/releases/download/1.7.5/Cryptomator-1.7.5-x64.msi
> 
> Expected:    0f8f14174967dcd804f9311cdc684cab8c91e1f84e3f99c89edee8e44880e030
> 
> Actual:
> 
> ERROR
> 
> Please try again or create a new issue by using the following link and paste your console output:
> 
> https://github.com/ScoopInstaller/Extras/issues/new?title=cryptomator%401.7.5%3a+hash+check+failed
> 
> 
> 
> 'cryptomator' (1.7.5) was downloaded successfully!
> 
> 
> 
> ```
> 
> 

What is the output of "echo $env:SCOOP"?

**Reply by Xelbayria** [reply]
It say `C:\Program Scoop`

the setup on my computer and my laptop are exactly the same

**Comment by Xelbayria** [comment]
I copied the `cryptomator.json` to my own bucket, renamed it `cryptomator-t.json`  so "t" for test. 

what I change is from this: 
```JSON
    "architecture": {
        "64bit": {
            "url": "https://github.com/cryptomator/cryptomator/releases/download/1.7.5/Cryptomator-1.7.5-x64.msi",
            "hash": "0f8f14174967dcd804f9311cdc684cab8c91e1f84e3f99c89edee8e44880e030"
        }
    },
```
to
Removing "architecture" & "64bit"  
```JSON
"url": "https://github.com/cryptomator/cryptomator/releases/download/1.7.5/Cryptomator-1.7.5-x64.msi",
"hash": "0f8f14174967dcd804f9311cdc684cab8c91e1f84e3f99c89edee8e44880e030"
```

The result is that it works without any problem. Are you able to figure out what's going on?

**Reply by ivaquero** [reply]
No, it's the first time for me to know this error.

**Comment by Xelbayria** [comment]
Ok. I've copied the json file and removed the architecture, now I am facing another problem: 
```shell
Resetting cryptomator-noarch (1.7.5).
Linking C:\Program Scoop\apps\cryptomator-noarch\current => C:\Program Scoop\apps\cryptomator-noarch\1.7.5
Creating shim for 'Cryptomator'.
OperationStopped: C:\Program Scoop\apps\scoop\current\lib\shortcuts.ps1:57
Line |
  57 |      $wsShell.Save()
     |      ~~~~~~~~~~~~~~~
     | Unable to save shortcut "C:\Users\Ronrad\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Scoop
     | Apps\Cryptomator.lnk".
Creating shortcut for Cryptomator (Cryptomator.exe)
Persisting data
```
😩 the result is the same thing as when I installed the software. I wasn't able to find out what went wrong.  The only thing I did is removed "architecture". just added "url" and "hash". that's it.

**Comment by HUMORCE** [comment]
```
> scoop install cryptomator
Installing 'cryptomator' (1.9.1) [64bit] from extras bucket
Loading Cryptomator-1.9.1-x64.msi from cache
Checking hash of Cryptomator-1.9.1-x64.msi ... ok.
Extracting Cryptomator-1.9.1-x64.msi ... done.
Linking ~\scoop\apps\cryptomator\current => ~\scoop\apps\cryptomator\1.9.1
Creating shim for 'Cryptomator'.
Creating shortcut for Cryptomator (Cryptomator.exe)
Persisting data
Running post_install script...
'cryptomator' (1.9.1) was installed successfully!
'cryptomator' suggests installing 'nonportable/winfsp-np'.
```
- hash: Cannot reproduce, unless it is remove by defender.
- shortcut: I guess, your env variable: `$env:SCOOP = 'C:\Program Scoop'`. try install with admin right.

**Reply by Xelbayria** [reply]
Alright, I've ran another test and it's working fine now. Not sure why. 

btw, I also ran with admin right and it also failed with the same error (talking about the last time).

**Comment by Xelbayria** [comment]
I'll be closing this discussion as it's no longer relevant. The issue have been solved somehow. Have a good day!
